@workflow/next fails to resolve next/package.json in npm workspaces monorepo

## Bug Report

### Environment
- `workflow`: 4.2.0 (GA) — also reproduces with 4.2.0-beta.78
- `@workflow/next`: 4.0.1
- `next`: 16.2.3
- Node.js: v24.13.1
- npm: 11.12.1
- Monorepo: npm workspaces (Turborepo)

### Description

`@workflow/next@4.0.1` does `require('next/package.json')` in `dist/index.js` (line references `fallbackVersion`). In an npm workspaces monorepo where `next` is declared as a dependency in a workspace package (e.g., `apps/web/package.json`), npm does not hoist `next` to the root `node_modules` — it stays in `apps/web/node_modules/next`. This is because Next.js uses a `require-hook` that affects hoisting behavior.

Since `@workflow/next` IS hoisted to the root `node_modules` (it's a transitive dependency of `workflow`), it can't resolve `next/package.json` from there.

### Reproduction

```
monorepo/
├── package.json          # workspaces: ["apps/web"]
├── node_modules/
│   └── @workflow/next/   # ← hoisted here, does require('next/package.json')
└── apps/web/
    ├── package.json      # depends on next, workflow
    └── node_modules/
        └── next/         # ← next lives here, NOT at root
```

```
Error: Cannot find module 'next/package.json'
Require stack:
- /monorepo/node_modules/@workflow/next/dist/index.js
- /monorepo/node_modules/workflow/dist/next.cjs
```

### Failing code

```js
// node_modules/@workflow/next/dist/index.js
const fallbackVersion = require('next/package.json').version;
```

### Expected behavior

`@workflow/next` should resolve `next` relative to the consuming project, not relative to its own install location. This could be done with `require.resolve('next/package.json', { paths: [process.cwd()] })` or by making the version check lazy/optional.

### Workaround

Stay on `workflow@^4.2.0-beta.76` which pins a `@workflow/next` version where the module resolution worked (likely due to different lock file hoisting).

--- a/.changeset/fix-next-version-resolution.md
+++ b/.changeset/fix-next-version-resolution.md
@@ -1,0 +1,5 @@
+---
+"@workflow/next": patch
+---
+
+Fix `next/package.json` resolution failure in npm workspaces monorepos

--- a/packages/next/src/index.ts
+++ b/packages/next/src/index.ts
@@ -7,6 +7,8 @@ import {
 } from './builder.js';
 
 function resolveNextVersion(workingDir: string): string {
+  const errors: unknown[] = [];
+
   // Try resolving from the consuming project's working directory first.
   // This handles monorepo setups where `next` may not be hoisted to the
   // same location as `@workflow/next`.
@@ -20,8 +22,8 @@ function resolveNextVersion(workingDir: string): string {
     if (typeof resolvedPackageJson.version === 'string') {
       return resolvedPackageJson.version;
     }
-  } catch {
-    // next/package.json not resolvable from working directory
+  } catch (e) {
+    errors.push(e);
   }
 
   // Fall back to resolving relative to this package's location.
@@ -31,12 +33,13 @@ function resolveNextVersion(workingDir: string): string {
     if (typeof version === 'string') {
       return version;
     }
-  } catch {
-    // next/package.json not resolvable from this package's location either
+  } catch (e) {
+    errors.push(e);
   }
 
-  throw new Error(
-    'Could not resolve Next.js version. Ensure `next` is installed in your project.'
+  throw new AggregateError(
+    errors,
+    `Could not resolve Next.js version. Ensure \`next\` is installed in your project (working directory: ${workingDir}).`
   );
 }
 

--- a/packages/next/src/index.ts
+++ b/packages/next/src/index.ts
@@ -7,8 +7,9 @@ import {
 } from './builder.js';
 
 function resolveNextVersion(workingDir: string): string {
-  const fallbackVersion = require('next/package.json').version as string;
-
+  // Try resolving from the consuming project's working directory first.
+  // This handles monorepo setups where `next` may not be hoisted to the
+  // same location as `@workflow/next`.
   try {
     const packageJsonPath = require.resolve('next/package.json', {
       paths: [workingDir],
@@ -16,12 +17,27 @@ function resolveNextVersion(workingDir: string): string {
     const resolvedPackageJson = require(packageJsonPath) as {
       version?: unknown;
     };
-    return typeof resolvedPackageJson.version === 'string'
-      ? resolvedPackageJson.version
-      : fallbackVersion;
+    if (typeof resolvedPackageJson.version === 'string') {
+      return resolvedPackageJson.version;
+    }
   } catch {
-    return fallbackVersion;
+    // next/package.json not resolvable from working directory
   }
+
+  // Fall back to resolving relative to this package's location.
+  try {
+    const version = (require('next/package.json') as { version?: unknown })
+      .version;
+    if (typeof version === 'string') {
+      return version;
+    }
+  } catch {
+    // next/package.json not resolvable from this package's location either
+  }
+
+  throw new Error(
+    'Could not resolve Next.js version. Ensure `next` is installed in your project.'
+  );
 }
 
 export function withWorkflow(

